solidity/contracts/libs/hooks/GlobalHookMetadata.sol

```diff
@@ -0,0 +1,130 @@
+// SPDX-License-Identifier: MIT OR Apache-2.0
+pragma solidity >=0.8.0;
+
+/*@@@@@@@       @@@@@@@@@
+ @@@@@@@@@       @@@@@@@@@
+  @@@@@@@@@       @@@@@@@@@
+   @@@@@@@@@       @@@@@@@@@
+    @@@@@@@@@@@@@@@@@@@@@@@@@
+     @@@@@  HYPERLANE  @@@@@@@
+    @@@@@@@@@@@@@@@@@@@@@@@@@
+   @@@@@@@@@       @@@@@@@@@
+  @@@@@@@@@       @@@@@@@@@
+ @@@@@@@@@       @@@@@@@@@
+@@@@@@@@@       @@@@@@@@*/
+
+/**
+ * Format of metadata:
+ *
+ * [0:1] variant
+ * [1:33] msg.value
+ * [34:65] Gas limit for message (IGP)
+ * [66:85] Refund address for message (IGP)
+ * [86:] Custom metadata
+ */
+library GlobalHookMetadata {
+    uint8 private constant VARIANT_OFFSET = 0;
```

Comment:
IMO, I would 1 or 2 more bits to allow for more prefixes

Reply:
2^8 bits isn't enough?